internal/xds/translator/ratelimit.go

```diff
@@ -136,27 +136,8 @@ func patchRouteWithRateLimit(route *routev3.Route, irRoute *ir.HTTPRoute) error
 		return nil
 	}
 	global := irRoute.Traffic.RateLimit.Global
-	rateLimits, costSpecified := buildRouteRateLimits(irRoute.Name, global)
-	if costSpecified {
-		// PerRoute global rate limit configuration via typed_per_filter_config can have its own rate routev3.RateLimit that overrides the route level rate limits.
-		// Per-descriptor level hits_addend can only be configured there: https://github.com/envoyproxy/envoy/pull/37972
-		// vs the "legacy" core route-embedded rate limits doesn't support the feature due to the "technical debt".
-		//
-		// This branch is only reached when the response cost is specified which allows us to assume that
-		// users are using Envoy >= v1.33.0 which also supports the typed_per_filter_config.
```

Comment:
It's safe to use typed_per_filter_config for the "ordinay" ratelimit configuration, as the EG v1.3.0 will bump Envoy to v1.33.0, which supports this feature.

Reply:
> as EG v1.3.0 will bump Envoy to v1.33.0, which supports this feature.

This will break the existing EG cluster running with the old Envoy I guess? so are we suggest users upgrade envoy fleet first before upgrading EG?